### ISSUE TITLE
기계학습 소개

### ISSUE THREAD

**Issue by minji-o-j** [minji-o-j/Machine-Learning#1]

**Comment by minji-o-j** [comment]
# >인공지능, 기계학습, 심층학습의 관계
## 1. 인공지능, 기계학습, 심층학습 관계와 정의
### 인공지능
- 기계가 사람이나 동물의 지능이나 행동 을 모방하게 하는 기술
- 환경을 인지하고 __목표__ 를 성공적으로 달성할 확률을 최대화 하는 행동을 취한다.
  - 합리적 행위, 행동을 추구함 ->목적, 목표지향적 기술
  - '객체' 에 초점  

### 기계학습
- ⭐ __명시적인 규칙(프로그래밍) 없이__ 기계(컴퓨터)가 __데이터로부터 학습__ 하도록 하는 기술
  - ex) 스팸필터기:  "파격 혜택"이라는 단어가 들어갈 경우 스팸으로 분류 ->__명시적 지시: 파격 혜택__
  - 스팸 필터는 스팸 메일과 일반 메일의 샘플을 구분하여 둘을 구분할 수 있는 방법을 배우는 머신러닝 프로그램이다.
    - A,B,D: 스팸, C,E,F: 스팸 아님, 이 자료를 보고 기준을 알아서 정해줌 -->명시적 규칙X


### 심층학습
- __인공 신경망__ 에 기반한 기계학습의 한 분야로, 고차원 특징을 추출하는 깊은 구조를 사용하는 기술.
- raw input으로부터 higher level features를 점진적으로 추출한다. --> __계층적 구조__
  - ex) 영상처리: 낮은 층의 레이어 - 선 성분 식별, 높은 층의 레이어 - 숫자, 문자, 얼굴과 같은 사람이 인지하는 것과 관련된 개념 식별 (layer마다 역할이 다르다.)
- 최근 빅데이터와 GPU의 발전을 토대로 큰 이목이 집중됨

---

## 2. 간락한 역사
![image](https://user-images.githubusercontent.com/45448731/85196144-0c47bd00-b313-11ea-89f5-2aff74195cc6.png)
![image](https://user-images.githubusercontent.com/45448731/85196161-22ee1400-b313-11ea-88cb-41d7b141f2ec.png)

**Comment by minji-o-j** [comment]
# > 기계학습 소개
- AI 겨울은 언제오는걸까? : 기술적 이상과 현실의 충돌

## 1. 기계학습이란?
- **데이터로부터 학습**하도록 컴퓨터를 프로그래밍하는 과학 또는 기술
  - 데이터로부터 기계를 학습시키는 기술
- **명시적인 프로그래밍 없이** 컴퓨터가 학습하는 능력을 갖추게 하는 연구분야
  - 아서 사무엘(Arthur Samuel, 1959, 기계학습이란 용어 대중화)

### ex) 스팸 필터
- **훈련 세트(training set)**: 학습을 위해 시스템이 사용하는 샘플 `메일 1~n`
- **훈련 예제(training example)**: 각 훈련 데이터 `1, 2, 3 ... n`
- **작업 T**(Task): 하고자 하는 일 `주어진 메일이 스팸인지 구별하는 것`
- **경험 E**(Experience): 훈련 데이터(training data)
- **성능 P**(Performance): (예) 올바르게 분류된 메일의 비율 **(정확도, accuracy)**

```
Q. 위키피디아 백과 문서를 모두 내 컴퓨터에 다운로드하면 내 컴퓨터가 똑똑해질까?

A: X -->기계학습의 조건을 만족하지 않는다.
    이유: 작업(목적, Task)이 없다. -->성능 측정 기준 X (데이터는 있지만, 목적에 맞는 특정 데이터가 없다.)
```

## 2. 기계학습은 왜 필요한가?
### 스팸 필터 분류기
#### - 전통적 프로그래밍 접근 방법
  - 스팸 규칙이 점점 복잡, 길어진다.
  - 유지 보수가 어렵다.
  - 약간의 수정 -> 규칙 작성을 새로 해줘야한다.
    - ex) For U -> 4U 등..
#### - 기계학습 접근 방법
  - 어떤 단어 혹은 구절이 스팸 **판단에 적합한 기준 (특성, feature)**인지 자동으로 학습
  - 기존 시스템으로는 **여러 수동 조절과 많은 규칙**이 필요한 문제
    - 스팸메일 진화에 따라 사람이 기준 계속 바꿔줌 VS 기준 자동으로 새로 생성
  - 유동적인 환경
  - 기존 방식으로는 해결 방법이 없는 **복잡한 문제**
  - 복잡한 문제와 **대량 데이터**에서 통찰 얻기

**Comment by minji-o-j** [comment]
## 3. 기계학습 시스템의 종류
### - 기계학습 시스템의 종류
- 지도 학습Supervised learning
- 준지도 학습Semi-supervised learning
- 비지도 학습Unsupervised learning

### - 구분 기준
- 학습 데이터에 레이블이 있는지
  - ex) 스팸메일 -> 레이블 S(스팸)or H(Ham, 스팸X)
- 학습하는 동안의 감독 형태나 정보량
  - 레이블이 다 있는지 / 일부만 있는지 / 없는지

---

### - 지도 학습 (Supervised learning)
- 훈련 데이터에 **레이블(label)** 이라는 구하고자 하는 출력 값이 포함됨
<br> 

- **분류** `ex: 스팸 필터`
  - 이산적인 값을 예측 (ex: 스팸인지 아닌지)
  - 이진분류 (o/x) 혹은 다중분류(A/B/C/D)  
<br> 

- **회귀** `ex: 중고차 가격 예측`
  - **연속적인 값**을 예측 (ex: 가격)
  - **특성**(feature) : 주행 거리, 연식 브랜드 등
  - 시스템 훈련을 위해 특성과 레이블(중고차 가격)이 필요

---

### - 비지도 학습 (Unsupervised Learning)
- 훈련 데이터에 **레이블이 없음**
  - ex: 학교에 들어오는 대학원생의 방문 시각, 체류시간에 따른 분포
  - ![image](https://user-images.githubusercontent.com/45448731/85197385-9562f200-b31b-11ea-9873-50d7ff795d39.png)
  - 전일 대학원생, 직장병행 대학원생: 레이블
<br>

- **군집화**clustering (예: 블로그 방문자)
  - 여러 그룹으로 나누고자 한다.
  - 키워드 '만화책', '저녁'에 방문, '남성' 다수 --> 하나의 그룹으로
  - 키워드 'SF', 주로 '주말'에 방문, '30대' --> 하나의 그룹으로
  - 집단 별 광고를 띄울 수 있다.
<br> 

- **시각화**visualization (예: 대규모 고차원 데이터 도식화)
  - 구별되지 않은 사건, 특징 1,2를 시각화
  - 정답이 아니며 단지 형태를 나타낸 것 뿐, 구분X
<br>

- **이상치 탐지**anomaly detection (예: 부정 신용 거래 방지)
  - 이상한 신용카드 거래 감지
  - 제조 상 결함 감지
  - 학습데이터에서 이상한 값을 자동으로 제거
  - ex) 00에서 로그인이 되었는데 본인이 맞습니까?
<br>

- **차원 축소**
  - 너무 많은 정보 손실을 피하면서 데이터를 간소화 (특성을 압축)
  - 예: 차의 주행 거리 + 연식 --> 차의 마모 정도 (특성 추출feature extraction)
<br>

- **연관 규칙 학습**association rule learning
  - 대용량 데이터에서 특성 간 유용한 관계를 찾음
  - 예: 편의점에서 컵라면을 구매한 고객은 삼각김밥도 구매하는 경향

---
### -준지도 학습 (Semi-supervised learning)
- 훈련 데이터에 **레이블이 일부만** 있음. (전체 데이터중 일부만 레이블이 되어있음)
- 지도 학습+ 비지도 학습
- ex) 사진 중 얼굴인식이 된 부분에 '나'임을 등록하면(레이블 표시), 다른 사진에서도 '나'를 찾아줌.

**Comment by minji-o-j** [comment]
## 4. 기계학습의 주요 도전과제
### - 기계학습을 어렵게 만드는 요인들
- 불량 데이터  

  - 부족한 양의 훈련 데이터
  - 대표성 없는 훈련 데이터(예: 나라별 행복도 예측을 위해 _아시아 지역 국가만_ 데이터를 수집) --> 치우쳐있음.
  - 관련 없는 특성(예: 행복도 예측에 키, 피부색 등은 부적합)
  - 저품질 데이터(에러, 센서불량으로 인한 잡음 등)

- 부적합한 알고리즘(5장 정규화)
  - 과대적합Overfitting 발생
  - 과소적합Underfitting 발생

**Comment by minji-o-j** [comment]
## 5. 테스트와 검증
- 데이터 세트를 두 개 또는 세 개의 그룹으로 나누어 기계학습 모델의 성능을 검증함

- **훈련 데이터** Training data

  - 모델을 훈련 시킬 때(학습시) 사용

  - 성능 측정시 절대 사용하면 안된다!!

- **테스트 데이터** Test data
  - 본 적 없는 새로운 샘플에 모델이 얼마나 잘 작동하는 지, 모델의 성능을 테스트 (성능 측정용)

- **검증 데이터** Validation data
  - 알고리즘을 훈련시킬 때, 여러 옵션 중 검증 데이터에 대해 **최상의 성능**을 내는 옵션을 선택 (튜닝용)

  - 알고리즘의 훈련 옵션: 하이퍼 파라미터Hyper parameter

  - validation set에서 하이퍼파라미터 조정-> 최상의 성능을 내는걸 test에서 사용
  - 테스트 세트는 최종 단 한 번의 테스트에서만 사용
